### PR TITLE
30 rsids discrepancies between studies

### PR DESCRIPTION
**Pull request by ritarasteiro**
This branch has updated LD harmonisation functions that use `variantids` instead of `rsids`.

* #36  branch was merged into this branch.
* merge this PR after #36 
* it passed RCMD checks without errors or warnings

* LD harmonisation functions
    * added `R/harmonise_ld.R:d_matrix_local()` based on `ieugwasr::ld_matrix_local()`. It queries plink bedfiles by chr:pos instead of `rsids` using the plink `--extract range` option.  The LD matrix row and colnames are the `variantids`, that are created by the `R/harmonise_ld.R:create_variantid_plink()`. 
    * changed `R/harmonise_ld.R:harmonise_ld_dat` to accept `variantids`
    * changes to `R/harmoniseLD_method.R:buildLDMatrix` to use `R/harmonise_ld.R:ld_matrix_local()`. The variants list needs to be in the [plink 'set range' format](https://www.cog-genomics.org/plink/1.9/input#make_set)
    *  changes to `R/harmoniseLD_method.R:harmoniseLDMatrix()` to accept `variantids`
    * updated `R/harmoniseLD_method.R:harmonise_ld()`

* `R/constructors.R`
    * `create_summaryset()` is now a function that calls the different `create_summaryset_from_?()`
    * the previous `create_summaryset()` is now called `create_summaryset_from_ tibble()`
    * new `add_summaryset()` function to add a summaryset to an existent dataset.

*  `R/meta.R`
    *  Christopher suggested to change the previous name `create_meta_summarysets()` to `meta_analysis` 

* small fixes, the majority to deal to the use of `variantids`

* updated vignettes

### CODE REVIEW

File: R/meta.R [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  trait <- paste(unique(traits), collapse = "||")
+
+  # eaf weighted mean
+  eaf <- lapply(1:length_dt, function(i) {
+    m <- getData(dataset,i)$eaf * getMetadata(getSummarySet(dataset,i))$sample_size
+    })
+  eaf_mean <- Reduce("+", eaf) / n_meta
+
+  # create metadata object
+  metadata <- create_metadata(metadata = NULL,
```
Comment (explodecomputer) [inline R/meta.R:79]: This assumes that those values are all present, but e.g. for this set of traits `chd_id <- c("ieu-a-7",  "ukb-d-I9_IHD", "finn-b-I9_CHD")` the last one has no sample size but it does have ncase and ncontrol, and so it errors. So the sample size is a sum of those two, but really the case control numbers are important too. Not completely sure what to do with this but perhaps at least would be helpful to be able to easily add sample size (or any metadata value) using a setter function

**Review by explodecomputer — COMMENTED** [review]
Hey thanks for this. A few suggestions -

File: R/harmonise_ld.R [hunk 2]
```diff
@@ -1,28 +1,94 @@
...
+	)
+	system(fun2)
+	res <- read.table(paste0(fn, ".ld"), header=FALSE) %>% as.matrix
+	rownames(res) <- colnames(res) <- paste(create_variantid_plink(bim), bim$V5, bim$V6, sep="|")
+	
+	return(res)
+}
+
+
+create_variantid_plink <-function(bim_file) {
```
[the remark this answers is not in the document]
Comment (ritarasteiro) [reply R/harmonise_ld.R:46]: The generic one is a S4 method that just applies to a S4 object. I could have a generic one, not a S4 method that we could use both on S4 objects and on the plink output

File: R/meta.R (hunk 1 above)
Comment (ritarasteiro) [reply R/meta.R:79]: When creating metadata for each summaryset the user can add the `sample_size`, and if none is given the `create_summaryset()` function tries to fill it using the `max($n)` from the summary statistics. In `meta.R`, there is an error if the `sample_size` is missing from one of the summarysets and advises to fill that value using `create_metadata()`. I could create a function like `add_to_metadata`?

File: R/harmonise_ld.R [hunk 3]
```diff
@@ -1,28 +1,94 @@
...
+#'
+#  LD matrix returns with variantid_ea_oa identifiers. Make sure that they are oriented to the same effect allele as the summary dataset. Summary dataset can be dat1 dataset or harmonised dartaset
 #
-# @param x harmonised dataset
-# @param ld Output from ld_matrix
-# @return List of dataset and harmonised LD matrix
+#' @param x harmonised dataset
+#' @param ld Output from ld_matrix
+#' @return List of dataset and harmonised LD matrix
 harmonise_ld_dat <- function(x, ld){
```
[the remark this answers is not in the document]
Comment (ritarasteiro) [reply R/harmonise_ld.R:78]: Yes, 1 needs to be done, I'll add this to the build matrix function. I'll check 2, but I think it is doing this.

File: R/meta.R (hunk 1 above)
Comment (ritarasteiro) [reply R/meta.R:79]: I changed the `SetMetadata()` and now we can easyly add to the Metadata

**Comment by ritarasteiro** [comment]
I'll merge this pull request, as it is a working version, but still be working on the ld harmonisation problem.

**Pull request merged**